packages/adminrouter/extra/src/nginx.master.conf

```diff
@@ -87,6 +87,9 @@ http {
         # Description: DC/OS GUI
         root /opt/mesosphere/active/dcos-ui/usr;
 
+        # prevent opening in iframe
+        add_header X-Frame-Options DENY;
```

Comment:
Correct me if I am wrong, but you are disabling it in ALL location block/endpoints. This also means for `/service` endpoint which may host users/clients services that may want to be embedded. Not to mention the things like `/acs` which is just an API endpoint - we should avoid adding headers and keep things simple/minimal.

Could you please identify the location blocks where it's necessary and narrow it down?